qa-tests-backend/src/test/groovy/ImageSignatureVerificationTest.groovy

```diff
@@ -491,34 +567,59 @@ nzTe7BpOmVwmqLkIefEJe5L4PSXtp2KFLZqGO/kY5A==
         return policyBuilder
     }
 
+    class CertificateVerificationArgs {
+        String chain
+        String identity
+        String issuer
+        Boolean ctlogEnabled
+    }
+
+    class TransparencyLogVerificationArgs {
+        Boolean enabled
+        String url
+    }
+
     // Helper to create a signature integration with given name, public keys, chain, identity, and issuer.
     private static String createSignatureIntegration(
-            String integrationName, Map<String, String> namedPublicKeys, String chain = "",
-            String identity = "", String issuer = "") {
+            String integrationName,
+            Map<String, String> namedPublicKeys,
+            CertificateVerificationArgs certVerification = null,
+            TransparencyLogVerificationArgs tlogVerification = null) {
         SignatureIntegration.Builder builder = SignatureIntegration.newBuilder()
-                .setName(integrationName)
+            .setName(integrationName)
 
         if (!namedPublicKeys.isEmpty()) {
             List<CosignPublicKeyVerification.PublicKey> publicKeys = namedPublicKeys.collect {
                 CosignPublicKeyVerification.PublicKey.newBuilder()
-                        .setName(it.key).setPublicKeyPemEnc(it.value)
-                        .build()
+                    .setName(it.key).setPublicKeyPemEnc(it.value)
+                    .build()
             }
             builder.setCosign(CosignPublicKeyVerification.newBuilder()
-                    .addAllPublicKeys(publicKeys)
-                    .build()
+                .addAllPublicKeys(publicKeys)
+                .build()
             )
         }
 
-        if (chain != "") {
+        if (certVerification?.identity && certVerification?.issuer) {
             CosignCertificateVerification verification = CosignCertificateVerification.newBuilder()
-                    .setCertificateChainPemEnc(chain)
-                    .setCertificateIdentity(identity)
-                    .setCertificateOidcIssuer(issuer)
+                .setCertificateChainPemEnc(certVerification?.chain ?: "")
+                .setCertificateIdentity(certVerification?.identity ?: "")
+                .setCertificateOidcIssuer(certVerification?.issuer ?: "")
+                .setCertificateTransparencyLog(CertificateTransparencyLogVerification.newBuilder()
+                    .setEnabled(certVerification?.ctlogEnabled ?: false)
                     .build()
+                )
+                .build()
+
             builder.addCosignCertificates(verification)
         }
 
+        builder.setTransparencyLog(TransparencyLogVerification.newBuilder()
+            .setEnabled(tlogVerification?.enabled ?: false)
+            .setUrl(tlogVerification?.url ?: "")
```

Comment:
maybe we can have a default value so it will never be null unless changed explicitly

Reply:
applies to above code as well